frog/responsive-images.rkt

```diff
@@ -0,0 +1,117 @@
+#lang racket/base
+
+(require net/url
+         racket/contract/base
+         racket/contract/region
+         racket/file
+         racket/function
+         racket/list
+         racket/path
+         racket/port
+         racket/system
+         racket/string
+         "params.rkt"
+         "util.rkt"
+         "verbosity.rkt"
+         (only-in srfi/1 list-index))
+
+(provide make-responsive clean-resized-images magick-available?)
+
+(define magick-available? (find-executable-path "magick"))
+
+(module+ test
+  (require rackunit))
+
+(define (image-width path)
+  (define (magick-width path)
+    (string->number
+     (with-output-to-string
+       (λ ()
+         ; Fails when running from inside DrRacket on Mac OS X due to pathologic PATH
+         (system (string-append "identify -format %w \"" (path->string path) "\""))))))
+  (define get-width magick-width)
+  (get-width path))
+
+(module+ test
+  (when magick-available?
+    (check-eq? (image-width (string->path "../example/img/800px-image.gif")) 800)))
+
+(define/contract (resize-image in new-width out)
+  (path? number? path? . -> . boolean?)
+  ;; Depend on ImageMagick
+  (define (magick-resize in w out)
+    (system (format "magick \"~a\" -geometry ~a \"~a\"" (path->string in) w (path->string out))))
+  (define resize magick-resize)
+  (prn1 "Resizing image ~a to ~a pixels... " (path->string in) new-width)
+  (if (> (current-verbosity) 1)
+      (time (resize in new-width out))
+      (resize in new-width out)))
+
+(module+ test
+  (when magick-available?
+    (define out (string->path "/tmp/resized.gif"))
```

Comment:
Instead of hardcoding `/tmp` it would be more portable to use `(find-system-path 'temp-dir)` -- or even `make-temporary-file` if ImageMagick will overwrite the file without complaint.

(Although 100% Windows compatibility isn't necessarily a goal, I do try to avoid making it harder in case someone wanted to volunteer to help do this someday.)